arm64-assembly-optimization.md

```diff
@@ -698,17 +698,90 @@ Another way to use efficient instructions is to consult the [software
 optimization guide](README.md#building-for-graviton2-graviton3-and-graviton3e).
 Many different combinations of instructions can accomplish the same result, and
 some are obviously better than others. Some instructions, like the
-absolute-difference-accumulate-long instructions (`SABAL` and `UABAL`) can only
+absolute-difference-accumulate-long instructions (`sabal` and `uabal`) can only
 be executed on half of the vector pipelines in Neoverse V1, which slashes the
 overall throughput of a series of independent instructions to two per cycle.
-Using instead a series of absolute-difference instructions (`SABD` or `UABD`),
-which can execute on all four pipelines, can increase the parallelism.  If there
+Using instead a series of absolute-difference instructions (`sabd` or `uabd`),
+which can execute on all four pipelines, can increase the parallelism. If there
 are enough of these instructions, this is worth the trade off of requiring a
 separate step to add up the accumulating absolute difference. There are other
 instructions which this applies to, so consult the software optimization guide
 to see what the throughput and latency numbers are for the instructions you are
 using.
 
+<details>
+<summary>
+Expand to see a code sample using the two different approaches for calculating
+sum-absolute-difference.
+</summary>
+The first function in this example uses `uabal` which is limited to half of the
+NEON execution units. The second uses more instructions to complete the final
+sum, but since the instructions can execute on all NEON execution units, it runs
+25% faster on Graviton3.
+
+```
+.text
+.global sad_asm_01
+.global sad_asm_02
+
+sad_asm_01:
+    movi    v17.2d, #0
+1:
+    ld1     {v0.16b-v3.16b}, [x0], #64
+    ld1     {v4.16b-v7.16b}, [x1], #64
+    uabdl   v16.8h, v0.8b,  v4.8b
```

Comment:
it would be good to add a brief comment on what each group of these instructions are doing.

Reply:
Yes, you are right. I will add comments.